app/standalone.go

```diff
@@ -41,6 +41,82 @@ type standaloneData struct {
 	installers               []installer.PayloadUpdatePerformer
 }
 
+func DoStandaloneDownload(device *dev.DeviceManager, updateURI string,
+	clientConfig client.Config, vKey []byte,
+	stateExec statescript.Executor) error {
+
+	var image io.ReadCloser
+	var imageSize int64
+	var err error
+	var upclient client.Updater
+
+	if strings.HasPrefix(updateURI, "http:") ||
+		strings.HasPrefix(updateURI, "https:") {
+		log.Infof("Performing remote update from: [%s].", updateURI)
+
+		var ac *client.ApiClient
+		// we are having remote update
+		ac, err = client.New(clientConfig)
+		if err != nil {
+			return errors.New("Can not initialize client for performing network update.")
+		}
+		upclient = client.NewUpdate()
+
+		log.Debug("Client initialized. Start downloading image.")
+
+		image, imageSize, err = upclient.FetchUpdate(ac, updateURI, 0)
+		log.Debugf("Image downloaded: %d [%v] [%v]", imageSize, image, err)
+	} else {
+		// perform update from local file
+		log.Infof("Start updating from local image file: [%s]", updateURI)
+		image, imageSize, err = installer.FetchUpdateFromFile(updateURI)
+
+		log.Debugf("Fetching update from file results: [%v], %d, %v", image, imageSize, err)
+	}
+
+	if image == nil || err != nil {
+		return errors.Wrapf(err, "Error while installing Artifact from command line")
+	}
+	defer image.Close()
+
+	fmt.Fprintf(os.Stdout, "Installing Artifact of size %d...\n", imageSize)
+	p := &utils.ProgressWriter{
+		Out: os.Stdout,
+		N:   imageSize,
+	}
+	tr := io.TeeReader(image, p)
+
+	standaloneData, err := doStandaloneInstallStatesDownload(ioutil.NopCloser(tr), vKey, device, stateExec)
+	if err != nil {
+		return err
+	}
+
+	err = storeStandaloneData(device.Store, standaloneData)
```

Comment:
This the main difference with the existing code. This is done directly after download and no more after the 'install'